knotx-core/knotx-template-engine/src/main/java/com/cognifide/knotx/engine/service/ServiceEngine.java

```diff
@@ -53,12 +56,18 @@ public ServiceEngine(Vertx vertx, TemplateEngineConfiguration serviceConfigurati
                     resp.subscribe(subscriber);
                     req.end();
                 });
-        return serviceResponse.flatMap(response ->
-                Observable.just(Buffer.buffer())
-                        .mergeWith(response.toObservable())
-                        .reduce(Buffer::appendBuffer))
-                .doOnNext(this::traceServiceCall)
-                .flatMap(buffer -> Observable.just(buffer.toJsonObject().getMap()));
+        return serviceResponse.flatMap(response -> Observable.just(Buffer.buffer())
```

Comment:
Please wrap
```
flatMap(response -> Observable.just(Buffer.buffer()).mergeWith(response.toObservable()).reduce(Buffer::appendBuffer))
```
into separate method, like collectBuffers or reduceBuffers. See HttpRepository#get and HttpRespository#reduceBuffers - they doing the same. It will give you more readable code here.